### PR TITLE
stm32,i2c: Fix large I2C transactions on I2C V1

### LINKED ISSUE #58866: STM32F4: Large I2C read/write access problem
On STM32F4, large (more than 256 bytes) read/write accesses fail when i try to communicate with an SE050 secure element from NXP.

Fix #43577 introduces a new mechanism which appears to not be implemented in I2C V1 devices like the STM32F4. Indeed I2C V1 devices seem to not implement `LL_I2C_EnableReloadMode` mechanism. Actually, in my opinion I2C V1 devices do not even require the use of any trick to send / receive data longer than 256 bytes.

On my side, i solved the problem by replacing the `i2c_stm32_transaction()` function by the following code:
```
static inline int
i2c_stm32_transaction(const struct device *dev,
		      struct i2c_msg msg, uint8_t *next_msg_flags,
		      uint16_t periph)
{
	int ret;
	if ((msg.flags & I2C_MSG_RW_MASK) == I2C_MSG_WRITE) {
		ret = stm32_i2c_msg_write(dev, &msg, next_msg_flags, periph);
	} else {
		ret = stm32_i2c_msg_read(dev, &msg, next_msg_flags, periph);
	}
	return ret;
}
```

### PR DESCRIPTION
**Pull request by ghost**
Previous commit #43577 added support of large transactions on I2C v2, this commit implements some changes to also add support of large transactions on I2C v1. Some refactoring is also done to put the code in the right source files.

Fixes zephyrproject-rtos#58866

### CODE REVIEW

File: drivers/i2c/i2c_ll_stm32_v2.c [hunk 1]
```diff
@@ -738,3 +738,52 @@ int stm32_i2c_configure_timing(const struct device *dev, uint32_t clock)
 
 	return 0;
 }
+
+int stm32_i2c_transaction(const struct device *dev,
+		      struct i2c_msg msg, uint8_t *next_msg_flags,
+		      uint16_t periph)
+{
+	/*
+	 * Perform a I2C transaction, while taking into account the STM32 I2C
```
Comment (erwango) [inline drivers/i2c/i2c_ll_stm32_v2.c:747]: To make sure people understand this differenciation was made on purpose, I propose to make it clear:
```suggestion
	 * Perform a I2C transaction, while taking into account the STM32 I2C V2
```

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for sharing this change!

I have not been able to test, some comments inline.
Also, please remove `stm32_i2c_msg_write/read` definitions from of `drivers/i2c/i2c_ll_stm32.h` and make them static in _v1 and _v2 files.

File: drivers/i2c/i2c_ll_stm32_v2.c [hunk 2]
```diff
@@ -738,3 +738,52 @@ int stm32_i2c_configure_timing(const struct device *dev, uint32_t clock)
 
 	return 0;
 }
+
+int stm32_i2c_transaction(const struct device *dev,
+		      struct i2c_msg msg, uint8_t *next_msg_flags,
+		      uint16_t periph)
```
[the remark this answers is not in the document]
Comment (trustngotech) [reply drivers/i2c/i2c_ll_stm32_v2.c]: done

**Comment by trustngotech** [comment]
Zephyr CI still complaining about coding style issues. Hope this time things will be ok...

**Comment by trustngotech** [comment]
Unfortunately, CI is still complaining about either indentation (tab vs. spaces etc.) or lines being to long. Don't want to fight with CI so indentation will stay as is because if this is not blocking.

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**